platform-api/src/main/java/io/datacater/core/deployment/K8Deployment.java

```diff
@@ -0,0 +1,247 @@
+  private final KubernetesClient client;
+  private final K8ConfigMap k8ConfigMap;
+  private final K8NameSpace k8NameSpace;
+
+  public K8Deployment(KubernetesClient client) {
+    this.client = client;
+    this.k8NameSpace = new K8NameSpace(client);
+    this.k8ConfigMap = new K8ConfigMap(client);
+  }
+
+  public String create(
+      DatacaterDeployment dcDeployment,
+      PipelineEntity pe,
+      StreamEntity streamIn,
+      StreamEntity streamOut) {
+    final String name = getName(dcDeployment.name());
+    final String volumeName = name + StaticConfig.VOLUME_NAME_SUFFIX;
+    k8NameSpace.create();
+
+    if (!pe.equals(new PipelineEntity())) {
+      k8ConfigMap.getOrCreate(name, pe);
+    }
+
+    if (exists(name)) {
+      return name;
+    }
+
+    List<EnvVar> variables = getEnvironmentVariables(dcDeployment, streamIn, streamOut);
+
+    var deployment =
+        new DeploymentBuilder()
+            .withNewMetadata()
+            .withName(name)
+            .addToLabels(
+                Map.of(
+                    StaticConfig.APP,
+                    StaticConfig.DATACATER_PIPELINE,
+                    StaticConfig.PIPELINE,
+                    StaticConfig.PIPELINE_NO,
+                    StaticConfig.REVISION,
+                    StaticConfig.PIPELINE_REV))
+            .endMetadata()
+            .withNewSpec()
+            .withReplicas(StaticConfig.REPLICAS)
+            .withMinReadySeconds(2)
+            .withNewSelector()
+            .addToMatchLabels(
+                Map.of(
+                    StaticConfig.APP,
+                    StaticConfig.DATACATER_PIPELINE,
+                    StaticConfig.PIPELINE,
+                    StaticConfig.PIPELINE_NO,
+                    StaticConfig.REVISION,
+                    StaticConfig.PIPELINE_REV))
+            .endSelector()
+            .withNewTemplate()
+            .withNewMetadata()
+            .addToLabels(
+                Map.of(
+                    StaticConfig.APP,
+                    StaticConfig.DATACATER_PIPELINE,
+                    StaticConfig.PIPELINE,
+                    StaticConfig.PIPELINE_NO,
+                    StaticConfig.REVISION,
+                    StaticConfig.PIPELINE_REV))
+            .endMetadata()
+            .withNewSpec()
+            .addNewContainer()
+            .withName(name)
+            .withImage(StaticConfig.EnvironmentVariables.FULL_IMAGE_NAME)
+            .withImagePullPolicy(StaticConfig.PULL_POLICY)
+            .withEnv(variables)
+            .withNewResources()
+            .withRequests(StaticConfig.RESOURCE_REQUESTS)
+            .withLimits(StaticConfig.RESOURCE_LIMITS)
+            .endResources()
+            .withVolumeMounts(
+                new VolumeMountBuilder()
+                    .withName(volumeName)
+                    .withMountPath(StaticConfig.MOUNT_PATH)
+                    .build())
+            .endContainer()
+            .withVolumes(
```

Comment:
What is the purpose of the volume?

Reply:
As far as I know, that's the standard when [mounting configMaps](https://kubernetes.io/docs/concepts/configuration/configmap/#using-configmaps).

Out manifest would look (mostly) like this:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: "2022-09-30T17:19:19Z"
  generation: 1
  labels:
    datacater.io/app: datacater-pipeline
    datacater.io/pipeline: "1"
    datacater.io/revision: "1"
  name: string3
  namespace: datacater
  resourceVersion: "1029646"
  uid: c9c2b8b5-da41-4d66-930d-e867bda91963
spec:
  minReadySeconds: 2
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      datacater.io/app: datacater-pipeline
      datacater.io/pipeline: "1"
      datacater.io/revision: "1"
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        datacater.io/app: datacater-pipeline
        datacater.io/pipeline: "1"
        datacater.io/revision: "1"
    spec:
      containers:
      - env:
        - name: MP_MESSAGING_OUTGOING_PIPELINE-OUT_TOPIC
          value: string
        - name: MP_MESSAGING_INCOMING_PIPELINE-IN_TOPIC
          value: string
        image: ghcr.io/datacater/pipeline:1
        imagePullPolicy: IfNotPresent
        name: string3
        resources:
          limits:
            memory: 4Gi
          requests:
            cpu: 100m
            memory: 1536Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /usr/app
          name: string3volume
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - configMap:
          defaultMode: 420
          name: string3
        name: string3volume
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2022-09-30T17:19:22Z"
    lastUpdateTime: "2022-09-30T17:19:22Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: "2022-09-30T17:19:19Z"
    lastUpdateTime: "2022-09-30T17:19:22Z"
    message: ReplicaSet "string3-7475c8dfb9" has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 1
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1
```

Reply:
Thanks for the clarification.